tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/hci.h>
+#include "mesh_test.h"
+#include "mesh/adv.h"
+
+#define LOG_MODULE_NAME test_adv
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_INF);
+
+#define WAIT_TIME 60 /*seconds*/
+
+struct bt_mesh_test_adv {
+	uint8_t retr;
+	int64_t interval;
+};
+
+static struct bt_mesh_send_cb send_cb;
+static struct bt_mesh_test_adv *xmit_param;
+static struct k_sem observer_sem;
+static const char *txt_msg = "adv test";
+static const char *cb_msg = "cb test";
+static int64_t tx_timestamp;
+static int seq_checker;
+
+static void test_tx_init(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+}
+
+static void test_rx_init(void)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+}
+
+static void bt_init(void)
+{
+	int err = bt_enable(NULL);
+
+	if (err) {
+		FAIL("Bluetooth init failed (err %d)", err);
+		return;
```

Comment:
```suggestion
		FAIL("Bluetooth init failed (err %d)", err);
```
Nitpick: I think `return` is redundant after a `FAIL` call. Applies several places here

Reply:
yep, this function is the result of removing copy-paste inside tests. Thanks for highlighting it.